modules/earthengine-layers/src/earth-engine-layer.js

```diff
@@ -116,27 +164,36 @@ export default class EarthEngineLayer extends CompositeLayer {
           minZoom,
           maxCacheSize,
           maxCacheByteSize,
+          frame,
 
-          async getTileData({x, y, z}) {
-            const imageUrl = getTileUrl(x, y, z);
-            const image = await load(imageUrl, ImageLoader);
-            return image;
-          },
+          getTileData: options => this.getFilmstripTileData(options),
+          // async getTileData({x, y, z}) {
+          //   console.log(getTileUrl);
+          //   const imageUrl = getTileUrl(x, y, z);
+          //   const image = await load(imageUrl, ImageLoader);
+          //   return image;
+          // },
 
           renderSubLayers(props) {
-            const {data, tile} = props;
+            const {data, tile, frame} = props;
             const {west, south, east, north} = tile.bbox;
             const bounds = [west, south, east, north];
 
+            let image;
+            if (Array.isArray(data)) {
+              image = data[frame];
+            } else if (data) {
+              image = data.then(result => result && result[frame]);
+            }
+
             return (
-              data && [
-                new BitmapLayer(
-                  Object.assign(props, {
-                    image: data,
-                    bounds
-                  })
-                )
-              ]
+              data &&
+              new BitmapLayer(
+                Object.assign(props, {
```

Comment:
Can we write `{...props, image, bounds}` instead of `Object.assign`? I think the deck code base may still use object.assign but most other repos are using the more modern syntax.

It is alo cleaner but of course a bit less efficient not to modify props.

Reply:
I've been using `Object.assign` because deck.gl has been using it. Didn't look to see if spreading was supported in browsers (though I assume it should still transpile correctly.)